src/content/docs/docs/guides/vue.mdx

```diff
@@ -0,0 +1,755 @@
+
+Next, update the template to include a button for deleting todos:
+
+```vue title="App.vue" ins={7}
+<template>
+  ...
+
+  <button @click="toggleTodoCompletion(todo)">
+    Mark as {{ todo.completed ? "Incomplete" : "Completed" }}
+  </button>
+  <button @click="deleteTodo(todo.id)">Delete</button>
+
+  ...
+</template>
+```
+
+That code change will render a delete button for each todo.
+
+Here is the completed `App.vue` file:
+
+```vue title="App.vue"
+<script setup lang="ts">
+import { BazaarApp, arrayMirrorSubscribeListener } from "@bzr/bazaar";
+import { ref, computed, type Ref } from "vue";
+
+const bzr = new BazaarApp({
+  appId: import.meta.env.VITE_APP_ID || "test",
+  loginRedirectUri: import.meta.env.VITE_URL || window.location.href,
+  onApiConnectError: async (bzr) => bzr.logOut(),
+});
+
+const isLoggedIn = ref(false);
+
+type Todo = {
+  id: string;
+  text: string;
+  completed: boolean;
+};
+
+const todoText = ref("");
+const fetchedTodos: Ref<Todo[]> = ref([]);
+const todos = computed(() => [...fetchedTodos.value].reverse());
+
+const todosCollection = bzr.collection<Todo>("todos");
+
+function onSubmit() {
+  const unpersistedTodo = { text: todoText.value, completed: false };
+  todosCollection.insertOne(unpersistedTodo);
+  todoText.value = "";
+}
+
+function toggleTodoCompletion(todo: Todo) {
+  todosCollection.updateOne(todo.id, { completed: !todo.completed });
+}
+
+function deleteTodo(id: string) {
+  if (!window.confirm("Delete todo forever?")) return;
+  todosCollection.deleteOne(id);
+}
+
+async function doLoginActions() {
+  isLoggedIn.value = true;
+  todosCollection.subscribeAll(
+    {},
+    arrayMirrorSubscribeListener(fetchedTodos.value),
+  );
+}
+
+// Perform login actions if the user is already logged in on page load
+if (bzr.isLoggedIn()) {
+  doLoginActions();
+}
+
+// Set up handler to perform login actions immediately after a successful login
+bzr.onLogin(doLoginActions);
+</script>
+
+<template>
+  <div class="container">
+    <h1>Todo</h1>
+    <template v-if="!isLoggedIn">
+      <button @click="bzr.login">Log in</button>
+    </template>
+    <template v-else>
+      <button @click="bzr.logOut">Log out</button>
+
+      <form class="todo-form" @submit.prevent="onSubmit">
+        <input
+          type="text"
+          v-model="todoText"
+          placeholder="Write a todo..."
+          required
+        />
+        <button>Add</button>
+      </form>
+
+      <div class="todos">
+        <div class="card" v-for="todo in todos" :key="todo.id">
+          <div class="todo-text" :class="{ strikethrough: todo.completed }">
+            {{ todo.text }}
+          </div>
+          <button @click="toggleTodoCompletion(todo)">
+            Mark as {{ todo.completed ? "Incomplete" : "Completed" }}
+          </button>
+          <button @click="deleteTodo(todo.id)">Delete</button>
+        </div>
+      </div>
+    </template>
+  </div>
+</template>
+
+<style scoped></style>
+```
```

Comment:
I would not add this. Maybe add a note saying to see the finished app, visit this GitHub repo.

Reply:
I really like having the completed code. If a reader has any problems along the way, it's super useful for debugging, or any uncertainty about where code snippets and changes should go. Also, it's really relatively small.

And I'd like to avoid linking GitHub repos. Linking a GitHub means the post and repo need to stay in sync. And in my experience such repos look out date quickly. Many times I've seen "last updated 4 years ago" across all assets in the repo for the tutorial and felt the value of the tutorial diminished because of seeming out of date.

Reply:
Another reason I don't mind making the post longer. I believe the success of the learner is more important than the length of the post. Again, from personal experience, when I'm following a tutorial I'm usually waiting for that one line of code or command that should do something simple, but I spend a whole day trying to get working, often because of something silly, the author making assumptions about the reader's knowledge, or just something out of date. And I'm very impressed when I get through a tutorial smoothly, and remember that "this author or publication does good tutorials".

Reply:
Basically, you are complaining about the myriad of terrible tutorials out there. I agree. But instead of baking in failsafes, we should just make sure we don't write a terrible tutorial and keep stuff up to date. It is part of our docs and not a blog post, so our motivation to keep it up to date is certainly higher.

Even if it is not super long, it is still 100 lines, and aesthetically, it is terrible. How about a compromise: We leave it in, but it is in an expandable block that is minimized (if that is possible in .mdx files).

Reply:
Yes, but also I think part of writing a good tutorial, in this case, is including the full file. It's to help if a learner makes a mistake as opposed to us having mistakes in the content. (Irrelevant side note, I think it looks fine :p)

Compromise accepted. DaisyUI has a component. Just pushed the commit